user/store.go

```diff
@@ -571,6 +571,20 @@ func (s *Store) FindAllAuthSubjectsForUser(ctx context.Context, userID string) (
 	return result, nil
 }
 
+func (s *Store) FindAuthSubjectForUser(ctx context.Context, providerID, userID string) (*AuthSubject, error) {
+	var result AuthSubject
+	err := s.AuthSubjectsFunc(ctx, "", func(sub AuthSubject) error {
+		if sub.ProviderID == providerID {
+			result = sub
+		}
+		return nil
+	}, userID)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
```

Comment:
I think this is redundant, `AuthSubjectsFunc` already supports filtering by providerID and user

Reply:
I wanted to have a helper function that only returned 1 user so it'd be easier to use within other functions without needing to loop through the result

Reply:
changing this to return a User object from a given subjectID